pkg/cli/tail_test.go

```diff
@@ -230,3 +420,20 @@ func (p *fakePuller) Pull(ctx context.Context, filter string, maxNum int) ([]*lo
 	p.gotMaxNum = maxNum
 	return p.logEntries, p.injectErr
 }
+
+func (p *fakePuller) StreamPull(ctx context.Context, filter string, logCh chan<- *loggingpb.LogEntry) error {
+	p.gotFilter = filter
+	if p.injectErr != nil {
+		return p.injectErr
+	}
+	for _, v := range p.logEntries {
+		logCh <- v
+	}
+
+	// mock real streamPull's behavior, looping until canceled instead of using
+	// a for{} that loops forever, this is more efficient in test in
+	// TestStreamTailCommand, the contex time out is set to 2 seconds sleep 3
+	// seconds here make sure the server is still running before the time out.
+	time.Sleep(3 * time.Second)
```

Comment:
Since we cancel the context anyway, do we still need this?

Reply:
I checked the failed tests, it seems always 3 seconds apart, wonder if this causes that.

Reply:
It doesn't appears on my laptop, and now the list command also fails sometimes, I think trim the TimeStamp will help fix the flaky test.